src/python/gudhi/representations/vector_methods.py

```diff
@@ -379,43 +383,16 @@ def transform(self, X):
         if not self.is_fitted():
             raise NotFittedError("Not fitted.")
 
-        if not X:
-            X = [np.zeros((0, 2))]
-        
         N = len(X)
 
-        events = np.concatenate([pd.ravel(order="F") for pd in X], axis=0)
-        sorting = np.argsort(events)
-        offsets = np.zeros(1 + N, dtype=int)
-        for i in range(0, N):
-            offsets[i+1] = offsets[i] + 2*X[i].shape[0]
-        starts = offsets[0:N]
-        ends = offsets[1:N + 1] - 1
+        if sum([len(x) for x in X]) == 0:
 
-        bettis = [[0] for i in range(0, N)]
-
-        i = 0
-        for x in self.grid_:
-            while i < len(sorting) and events[sorting[i]] <= x:
-                j = np.searchsorted(ends, sorting[i])
-                delta = 1 if sorting[i] - starts[j] < len(X[j]) else -1
-                bettis[j][-1] += delta
-                i += 1
-            for k in range(0, N):
-                bettis[k].append(bettis[k][-1])
-
-        return np.array(bettis, dtype=int)[:, 0:-1]
-
-    def fit_transform(self, X, y = None):
-        """
-        The result is the same as fit(X) followed by transform(X), but potentially faster.
-        """
-
-        if self.predefined_grid is None and self.resolution is None:
-            if not X:
-                X = [np.zeros((0, 2))]
-
-            N = len(X)
+            if self.resolution is None:
+                return np.zeros((N, 1))
```

Comment:
I think in the exact case we can have `resolution is None` and still have a `grid_` (of length larger than 1).

Reply:
Yes, I added some code to define `self.grid_` in `transform` if the class has not been fitted yet.

Reply:
That's not what I meant.
```python
import numpy as np
from gudhi.representations import BettiCurve
bc=BettiCurve(resolution=None)
bc.fit([np.array([[1,2],[4,6],[9,12]])])
BettiCurve(resolution=None)
bc.grid_
```
```pre
array([-inf,   1.,   2.,   4.,   6.,   9.,  12.])
```
```python
bc.transform([np.array([[1,2],[4,6],[9,12]])])
```
```pre
array([[0, 1, 0, 1, 0, 1, 0]])
```
```python
bc.transform([])
```
```pre
Empty list or empty diagrams: output contains only zeros
array([], shape=(0, 1), dtype=float64)
```
I believe we want shape=(0, 7) here.
```python
e=np.zeros((0,2))
bc.transform([e,e])
```
```pre
Empty list or empty diagrams: output contains only zeros
array([[0.],
       [0.]])
```
```python
bc.transform([e,np.array([[1,2],[4,6],[9,12]])])
```
```pre
array([[0, 0, 0, 0, 0, 0, 0],
       [0, 1, 0, 1, 0, 1, 0]])
```
So the transform of `e` is not the same depending on the other diagrams in X, that looks wrong.

Let me post a suggestion in a separate comment.